lightning-endian.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/ccan/ccan/endian/endian.h b/ccan/ccan/endian/endian.h
+index 3753f490..6e9d8046 100644
+--- a/ccan/ccan/endian/endian.h
++++ b/ccan/ccan/endian/endian.h
+@@ -104,8 +104,12 @@ static inline uint64_t bswap_64(uint64_t val)
+ #endif
+
+ /* Needed for Glibc like endiness check */
++#ifndef __LITTLE_ENDIAN
+ #define	__LITTLE_ENDIAN	1234
++#endif
++#ifndef __BIG_ENDIAN
+ #define	__BIG_ENDIAN	4321
++#endif
```

Comment:
What is this change required for? It might need upstreaming as well, since we should be able to detect endiannes reliably.

Reply:
Good point we should try to give it a go without

Reply:
Configurator has an option about HAVE_BIG_ENDIAN,  but I am not sure solve the issue during building ccan https://github.com/ElementsProject/lightning/blob/359433f374472a6128993bbaace2c1c7c62ee107/ccan/tools/configurator/configurator.c#L174

Reply:
The issue imho is that the `configurator` is running on the host architecture and will thus use the host endianness instead of the target arch's endianness. Maybe https://github.com/ElementsProject/lightning/pull/3464 could be used to run the configurator in qemu emulating the target arch, thus picking up the correct values automatically?